src/lib/tasks/Render.ts

```diff
@@ -1,121 +1,93 @@
-import type { HTTPResponse } from 'puppeteer-core/lib/esm/puppeteer/api-docs-entry';
+import type { Response } from 'playwright';
 
-import { report } from 'helpers/errorReporting';
-import { stats } from 'helpers/stats';
 import { injectBaseHref } from 'lib/helpers/injectBaseHref';
 import type { RenderTaskParams } from 'lib/types';
 
 import { Task } from './Task';
 
 export class RenderTask extends Task<RenderTaskParams> {
   async process(): Promise<void> {
-    const { url, waitTime } = this.params;
-    const { page } = this.page;
+    if (!this.page) {
+      throw new Error('Calling process before createContext()');
+    }
+
+    /* Setup */
+    const { url } = this.params;
     const baseHref = url.origin;
+    const page = this.page.page!;
+    let response: Response;
 
-    const total = Date.now();
-    const minWait = waitTime!.min;
-    let start = Date.now();
+    // Important to catch any redirect
+    this.page.setDisableNavigation(url.href, async (newUrl) => {
+      await this.page?.saveMetrics();
+      this.page?.page?.close();
```

Comment:
looks like an await is missing here or a .catch(() => {}) should be used.

Reply:
indeed 👀